src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -89,19 +89,21 @@ const VideoInputBackgroundBlurControl: React.FC<Props> = ({
       try {
         // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
         // If background blur is not on, then do a normal video selection
+        let newDevice: VideoInputDevice = deviceId;
         if (isVideoTransformDevice(selectedDevice) && !isLoading) {
-          setIsLoading(true);
+          setIsLoading(false);
           if ('chooseNewInnerDevice' in selectedDevice) {
-            const transformedDevice =
-              // @ts-ignore
-              selectedDevice.chooseNewInnerDevice(deviceId);
-            await selectVideoInput(transformedDevice);
+            // @ts-ignore
+            newDevice = selectedDevice.chooseNewInnerDevice(deviceId);
           } else {
             logger.error('Transform device cannot choose new inner device');
           }
-          setIsLoading(false);
+        }
+
+        if (isVideoEnabled) {
+          await meetingManager.startVideoInputDevice(newDevice);
         } else {
-          await selectVideoInput(deviceId);
+          meetingManager.selectVideoInputDevice(newDevice);
```

Comment:
I thought we will be maintaining state locally in this component or we keep it now in meetingManager and publish an update just on selection?

Reply:
I got this, we initially decided to go with local UI component state but it seems moved to meetingManager now.

Reply:
Yes, discussed with Trung offline, moved to `meetingManager`. Forgot telling you. :p